packages/ts-invariant/src/invariant.ts

```diff
@@ -25,17 +25,29 @@ export function invariant(condition: any, message?: string | number) {
   }
 }
 
-function wrapConsoleMethod(method: "warn" | "error") {
+const verbosityLevels = ["log", "warn", "error", "silent"] as const;
+type VerbosityLevel = (typeof verbosityLevels)[number];
+type ConsoleMethodName = Exclude<VerbosityLevel, "silent">;
+let verbosityLevel = verbosityLevels.indexOf("log");
+
+function wrapConsoleMethod<M extends ConsoleMethodName>(method: M) {
   return function () {
-    return console[method].apply(console, arguments as any);
-  } as (...args: any[]) => void;
+    if (verbosityLevels.indexOf(method) >= verbosityLevel) {
+      return console[method].apply(console, arguments as any);
+    }
+  } as (typeof console)[M];
 }
 
 export namespace invariant {
+  export const log = wrapConsoleMethod("log");
```

Comment:
Support for `invariant.log` is new with this PR (we used to have only `invariant.warn` and `invariant.error`).